### PR TITLE
V2 and upgrade to v0.23

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by edhelas** [comment]
Ping on that :) !

File: scripts/install [hunk 1]
```diff
@@ -133,34 +61,33 @@ ynh_script_progression --message="Configuring NGINX web server..."
...
-#=================================================
-# SETUP SYSTEMD
-#=================================================
-ynh_script_progression --message="Configuring a systemd service..."
+ynh_add_config --template="../conf/db.example.inc.php" --destination="$install_dir/config/db.inc.php"
 
-# Create a dedicated systemd config
-ynh_add_systemd_config
+chmod 400 "$install_dir/config/db.inc.php"
+chown $app:$app "$install_dir/config/db.inc.php"
```
Comment (edhelas) [inline scripts/install]: This is now replaced by a unique .env file https://github.com/movim/movim/blob/master/INSTALL.md#2-dotenv-configuration :)

**Comment by ericgaspar** [comment]
!testme

File: scripts/install [hunk 2]
```diff
@@ -9,112 +9,33 @@
...
-is_public=$YNH_APP_ARG_IS_PUBLIC
-admin=$YNH_APP_ARG_ADMIN
-password=$YNH_APP_ARG_PASSWORD
-
-app=$YNH_APP_INSTANCE_NAME
-
 timezone=$(cat /etc/timezone)
 fpm_footprint="low"
 fpm_free_footprint=0
 fpm_usage="low"
```
Comment (alexAubin) [inline scripts/install]:
```suggestion
```

File: scripts/install [hunk 3]
```diff
@@ -9,112 +9,33 @@
...
 # STORE SETTINGS FROM MANIFEST
 #=================================================
-ynh_script_progression --message="Storing installation settings..."
 
-ynh_app_setting_set --app=$app --key=domain --value=$domain
-ynh_app_setting_set --app=$app --key=path --value=$path_url
-ynh_app_setting_set --app=$app --key=admin --value=$admin
 ynh_app_setting_set --app=$app --key=fpm_footprint --value=$fpm_footprint
 ynh_app_setting_set --app=$app --key=fpm_free_footprint --value=$fpm_free_footprint
 ynh_app_setting_set --app=$app --key=fpm_usage --value=$fpm_usage
```
Comment (alexAubin) [inline scripts/install]:
```suggestion
```

File: scripts/install [hunk 4]
```diff
@@ -123,53 +44,37 @@ ynh_script_progression --message="Configuring PHP-FPM..."
 
 # Create a dedicated PHP-FPM config
 ynh_add_fpm_config --usage=$fpm_usage --footprint=$fpm_footprint
```
Comment (alexAubin) [inline scripts/install]:
```suggestion
ynh_add_fpm_config --usage=low --footprint=low
```

File: scripts/install [hunk 5]
```diff
@@ -123,53 +44,37 @@ ynh_script_progression --message="Configuring PHP-FPM..."
...
-#=================================================
-# GENERIC FINALIZATION
-#=================================================
-# INTEGRATE SERVICE IN YUNOHOST
-#=================================================
-ynh_script_progression --message="Integrating service in YunoHost..."
-
-yunohost service add $app --description="Responsive web-based XMPP client"
+#ynh_exec_warn_less ynh_exec_as "$app" ynh_composer_exec --commands="install"
+ynh_exec_warn_less ynh_exec_as "$app" ynh_composer_exec --commands="movim:migrate"
```
Comment (alexAubin) [inline scripts/install]:
```suggestion
#ynh_exec_warn_less ynh_exec_as "$app" ynh_composer_exec --commands="movim:migrate"
ynh_composer_exec --commands="movim:migrate"
```

File: scripts/install [hunk 6]
```diff
@@ -123,53 +44,37 @@ ynh_script_progression --message="Configuring PHP-FPM..."
...
+chown $app:$app "$install_dir/.env"
 
 #=================================================
 # BUILD MOVIM
 #=================================================
 ynh_script_progression --message="Building Movim..."
 
-ynh_exec_warn_less ynh_install_composer --phpversion="$phpversion" --workdir="$final_path"
-ynh_exec_warn_less ynh_composer_exec --phpversion="$phpversion" --workdir="$final_path" --commands="movim:migrate"
-ynh_exec_as $app php$phpversion $final_path/daemon.php config --username=$admin --password=$password --quiet
```
Comment (alexAubin) [inline scripts/install]: uuh that thing seems to create the admin user, seems important ?

**Comment by alexAubin** [comment]
!testme

File: scripts/install (hunk 6 above)
Comment (ericgaspar) [reply scripts/install]: I had trouble updating this app. I think this command is now obsolete but I will have to dig deeper to remember how to generate the admin (maybe a post install process)

**Comment by Salamandar** [comment]
!testme

**Comment by Salamandar** [comment]
I splitted the PR into packaging v2 (merged to testing) and all those other changes.

**Comment by ericgaspar** [comment]
!testme

**Comment by OniriCorpe** [comment]
!testme

**Comment by OniriCorpe** [comment]
> 24390 INFO DEBUG - Do not run Composer as root/super user! See https://getcomposer.org/root for details
> 24390 INFO DEBUG - Aborting as no plugin should be loaded if running as super user is not explicitly allowed

**Pull request closed without merging**